sphinx/events.py

```diff
@@ -64,8 +171,10 @@ def add(self, name: str) -> None:
             raise ExtensionError(__('Event %r already present') % name)
         self.events[name] = ''
 
-    def connect(self, name: str, callback: Callable, priority: int) -> int:
+    def connect(self, name: str | CoreEvent, callback: Callable, priority: int) -> int:
```

Comment:
we could take this further and make the `CoreEvent` enum the preferred approach (without breaking existing code)

```suggestion
    @deprecated(reason="...")
    @overload
    def connect(self, name: str, callback: Callable, priority: int) -> int: ...
    @overload
    def connect(self, name: CoreEvent, callback: Callable, priority: int) -> int: ...
    def connect(self, name: str | CoreEvent, callback: Callable, priority: int) -> int:
```

Reply:
The deprecated decorator is only a 3.13 feature :(

Reply:
(In addition, you still want to connect with your own events)